File: algo/62. Unique Paths.py
# Dynamic Programming
class Solution:
    def uniquePaths(self, m: 'int', n: 'int') -> 'int':
        helper_matrix = [[0]*n]*m
        for i in range(m):
            for j in range(n):
                if i-1>=0 and j-1>=0:
                    helper_matrix[i][j] = helper_matrix[i-1][j] + helper_matrix[i][j-1]
                elif i-1>=0:
                    helper_matrix[i][j] = helper_matrix[i-1][j]
                elif j-1>=0:
                    helper_matrix[i][j] = helper_matrix[i][j-1]
                else:   # the start
                    helper_matrix[i][j] = 1
        return helper_matrix[m-1][n-1]

File: round3/62. Unique Paths.cs
public class Solution {
    public int UniquePaths(int m, int n) {
        var dp = new int[n];
        
        for (int j = 0; j < n; j ++) {
            dp[j] = 1;
        }
        
        for (int i = 1; i < m; i ++) {
            for (int j = 1; j < n; j ++) {
                dp[j] = dp[j] + dp[j-1];
            }
        }
        
        return dp[n-1];
    }
}